jenkins/coverage/test.sh

```diff
@@ -0,0 +1,6 @@
+# exit on failure of any test line
+set -x
+set -e
+# Modify test/regression to add tests
+docker run -u $(id -u ${USER}):$(id -g ${USER}) -v $(pwd):/OpenROAD openroad/openroad bash -c "/OpenROAD/test/regression"
+docker run -v $(pwd):/OpenROAD openroad/openroad bash -c "yum install -y http://downloads.sourceforge.net/ltp/lcov-1.14-1.noarch.rpm && lcov --capture --directory /OpenROAD/build --exclude '/usr/include/*' --exclude '/opt/*' --exclude '/usr/lib/*' --exclude '/usr/local/*' --exclude '*build*' --output-file /OpenROAD/main_coverage.info && genhtml /OpenROAD/main_coverage.info --output-directory /OpenROAD/out --ignore-errors source"
```

Comment:
I don't agree with installing this to the container instead of a [multi-stage image](https://docs.docker.com/develop/develop-images/multistage-build) that inherits from the current Dockerfile. There are big problems with this:
1. Docker cannot cache the install and will unnecessarily reinstall every time.
2. The containers don't have `--rm` and will bog down the Docker cache. The pipeline doesn't have a cleanup step either.

Reply:
So, you are suggesting to create a new stage in the existing dockerfile that inherits from base-dependencies where installing lcov takes place,and to add a cleanup stage to remove the generated files for coverage report?

Reply:
That sounds about right. This would make the image with lcov cacheable. It doesn't need a cleanup stage if you use `docker run --rm` to remove the container after running.

The bash line is also very long. Is there some way to reformat it to be shorter and more readable?